### ISSUE TITLE
Incongruity with the `Context` type across the library.

### ISSUE THREAD

**Issue by Byloth** [strawberry-graphql/strawberry#3264]
I'm developing an app using Django + Strawberry + Channels.

Until the introduction of Channels and related logic to manage subscriptions, I used a custom decorator to populate the `info.context` object of some selected queries and mutations with some extra stuff.

To be clear, something like this:
```py
from functools import wraps

def my_decorator(funct):

    @wraps(funct)
    def my_wrapper(info, *args, **kwargs):
        info.context.my_property = get_some_extra_property()

        return funct(info, *args, **kwargs)

    return my_wrapper
```

 It worked fine.

---

Now...
After introducing support for subscriptions, I ran into an exception: it seems that the `context` instantiated by [`GraphQLView`](https://github.com/strawberry-graphql/strawberry/blob/a90dea2a3053e857aa2adf685d5c4f5d39c527c4/strawberry/django/views.py#L198-L199) is actually different from the one instantiated by [`GraphQLHTTPConsumer`](https://github.com/strawberry-graphql/strawberry/blob/main/strawberry/channels/handlers/http_handler.py#L239-L245)... Even though they both only expose 2 properties: `request` & `response`.  
This led the application to raise a:

```
AttributeError: 'dict' object has no attribute 'my_property'.
```

---

To be honest, I've already fixed my application so that's ok but this difference triggers my OCDs, IMHO isn't really a good thing for a library and it creates user confusion and that's why I opened this Issue.  
Make your own consideration and choose whatever approach you like... It's just a simple report.

**Comment by Byloth** [comment]
This issue is probably also related to:

- #2687.

**Comment by patrick91** [comment]
yup, we really should do #2687, that would make things quite a bit simpler for users I think
